packages/nx-bun/src/executors/run/executor.ts

```diff
@@ -1,126 +1,309 @@
+import { parentPort } from 'node:worker_threads';
+import * as path from "node:path";
+import {ChildProcess, spawn} from "node:child_process";
+import {randomUUID} from "node:crypto";
 import { createAsyncIterable } from '@nx/devkit/src/utils/async-iterable';
+import {
+  ExecutorContext,
+  parseTargetString,
+  ProjectGraphProjectNode,
+  readTargetOptions,
+  logger,
+  runExecutor
+} from '@nx/devkit';
+import {getRelativeDirectoryToProjectRoot} from "@nx/js/src/utils/get-main-file-dir";
+
+import {assertBunAvailable, executeCliAsync, isBunSubprocess, UnifiedChildProcess} from '../../utils/bun-cli';
+import {killCurrentProcess, Signal} from '../../utils/kill';
 import { RunExecutorSchema } from './schema';
-import { assertBunAvailable, executeCliAsync, isBunSubprocess } from '../../utils/bun-cli';
-import { parentPort, } from 'worker_threads';
-import { killCurrentProcess } from '../../utils/kill';
-import { ExecutorContext } from '@nx/devkit';
+import {debounce} from "../../utils/debounce";
+
+
+interface ActiveTask {
+  id: string;
+  killed: boolean;
+  promise: Promise<void> | null;
+  childProcess: UnifiedChildProcess | null;
+  start: () => Promise<void>;
+  stop: (signal: Signal) => Promise<void>;
+}
 
+function getFileToRun(
+  context: ExecutorContext,
+  project: ProjectGraphProjectNode,
+  buildOptions: Record<string, any>,
+  buildTargetExecutor: string
+): string {
+  // If using run-commands or another custom executor, then user should set
+  // outputFileName, but we can try the default value that we use.
+  if (!buildOptions?.outputPath && !buildOptions?.outputFileName) {
+    const fallbackFile = path.join('dist', project.data.root, 'main.js');
+    logger.warn(
+      `Build option outputFileName not set for ${project.name}. Using fallback value of ${fallbackFile}.`
+    );
+    return path.join(context.root, fallbackFile);
+  }
+
+  let outputFileName = buildOptions.outputFileName;
 
-export default async function* runExecutor(options: RunExecutorSchema, context: ExecutorContext) {
+  if (!outputFileName) {
+    const fileName = `${path.parse(buildOptions.main).name}.js`;
+    if (
+      buildTargetExecutor === '@nx/js:tsc' ||
+      buildTargetExecutor === '@nx/js:swc' ||
+      buildTargetExecutor === '@nx-bun/nx:build'
```

Comment:
I was thinking if the target is `nx-bun/nx` just to ignore this and use the run command directly? Spawning a new process to build would cause performance loss as bun has typescript build in. I do agree with tsc and swc. Once the nx/js is fully integrated, I was adding builder option too for vite and esbuild

Reply:
@Jordan-Hall https://github.com/Jordan-Hall/nx-bun/pull/2/files#diff-fa25683b3d2202a802e07a14a5d6a4bec89e83bf5448df5c40bdd60f3e6c7604R238 already ahead of you

Reply:
O yea :) Let me look tomorrow with clear head its almost 2am here. Are you able to rebase this branch to resolve the conflicts please